rcl/src/rcl/publisher.c

```diff
@@ -124,46 +124,47 @@ rcl_publisher_get_default_options()
 rcl_ret_t
 rcl_publish(const rcl_publisher_t * publisher, const void * ros_message)
 {
-  RCL_CHECK_ARGUMENT_FOR_NULL(publisher, RCL_RET_INVALID_ARGUMENT);
-  RCL_CHECK_ARGUMENT_FOR_NULL(ros_message, RCL_RET_INVALID_ARGUMENT);
-  RCL_CHECK_FOR_NULL_WITH_MSG(
-    publisher->impl, "publisher is invalid", return RCL_RET_PUBLISHER_INVALID);
-  if (rmw_publish(publisher->impl->rmw_handle, ros_message) != RMW_RET_OK) {
+  if (!rcl_publisher_is_valid(publisher)) return RCL_RET_PUBLISHER_INVALID;
+  else if (rmw_publish(publisher->impl->rmw_handle, ros_message) != RMW_RET_OK) {
     RCL_SET_ERROR_MSG(rmw_get_error_string_safe());
     return RCL_RET_ERROR;
-  }
-  return RCL_RET_OK;
+  } return RCL_RET_OK;
 }
 
 const char *
 rcl_publisher_get_topic_name(const rcl_publisher_t * publisher)
 {
-  RCL_CHECK_ARGUMENT_FOR_NULL(publisher, NULL);
-  RCL_CHECK_FOR_NULL_WITH_MSG(
-    publisher->impl, "publisher is invalid", return NULL);
-  RCL_CHECK_FOR_NULL_WITH_MSG(
-    publisher->impl->rmw_handle, "publisher is invalid", return NULL);
+  if (!rcl_publisher_is_valid(publisher)) return NULL;
```

Comment:
Ok -- I just removed the extra check. Now it just checks that `publisher->impl` is not `NULL`. I also changed it to return `false` instead of `NULL`, because I'm not sure why I had it as `NULL` instead of `false`.